alibi_detect/saving/schemas.py

```diff
@@ -24,24 +24,23 @@
 from alibi_detect.cd.tensorflow import UAE as UAE_tf
 from alibi_detect.cd.tensorflow import HiddenOutput as HiddenOutput_tf
 from alibi_detect.utils._types import Literal, NDArray
-from alibi_detect.utils.frameworks import has_tensorflow
+from alibi_detect.utils.frameworks import has_tensorflow, has_pytorch, has_sklearn
 from alibi_detect.version import __config_spec__, __version__
 
-SupportedModels_list = []
+# Define supported models for each optional dependency
+SupportedModels_tf, SupportedModels_torch, SupportedModels_sklearn = (), (), ()  # type: ignore
 if has_tensorflow:
     import tensorflow as tf
-    SupportedModels_list += [tf.keras.Model, UAE_tf, HiddenOutput_tf]
-# if has_pytorch:
-#    import torch
-#    SupportedModels_list.append()  # TODO
-# if has_sklearn:
-#    import sklearn
-#    SupportedModels_list.append()  # TODO
-# SupportedModels is a tuple of possible models (conditional on installed deps). This is used in isinstance() etc.
-SupportedModels = tuple(SupportedModels_list)
-# SupportedModels_types is a typing Union for use with pydantic. We include all optional deps in here so that they
-# are all documented in the api docs (where the optional deps are not installed at build time)
-SupportedModels_types = Union['tf.keras.Model', UAE_tf, HiddenOutput_tf]
+    SupportedModels_tf = (tf.keras.Model, UAE_tf, HiddenOutput_tf)  # type: ignore
+if has_pytorch:
+    # import torch
```

Comment:
this raised a fake error which is why I've commented it out

Reply:
Yep good idea 👍🏻